modules/gojira-core/src/main/java/com/flipkart/gojira/core/ProfileRepository.java

```diff
@@ -98,13 +117,36 @@ public static void setTestDataId(String testDataId) {
   }
 
   static synchronized Mode getMode() {
+    if (Mode.DYNAMIC.equals(globalProfileSetting.getMode())) {
+      return threadLocalProfileSetting.get().getMode();
+    }
     return globalProfileSetting.getMode();
   }
 
   static synchronized void setMode(Mode mode) {
     globalProfileSetting.setMode(mode);
   }
 
+  static synchronized void setRequestMode(String modeHeader) {
+    ProfileSetting profileSetting = new ProfileSetting();
+
+    if (Mode.DYNAMIC.equals(globalProfileSetting.getMode())) {
+      try {
+        if (StringUtils.isEmpty(modeHeader) || Mode.DYNAMIC.name().equals(modeHeader)) {
```

Comment:
use null check and String.isEmpty() as StringUtils is not extensively required here. And commons-lang3 dependecy can be removed.

Reply:
Done